src/sql/workbench/browser/modelComponents/listView.component.ts

```diff
@@ -68,6 +68,9 @@ export default class ListViewComponent extends ComponentBase<azdata.ListViewComp
 				this._optionsList.setSelection([this._optionsList.getFocus()[0]]);
 				DOM.EventHelper.stop(e, true);
 			}
+			if (e.keyCode === KeyCode.Escape) {
+				(this._optionsList.getHTMLElement() as HTMLElement).blur();
```

Comment:
the dialog is not getting closed when Escape key is pressed is because the escape key press is handled by the list component or its base classes, have you checked this? if it is the case, then we don't have to explicitly do blur here.